gcode/cmds.go

```diff
@@ -97,23 +112,25 @@ func retractionToGCode(b *bytes.Buffer, retraction bool, retractionDistance floa
 	b.WriteString("G1 F" + strconv.Itoa(retractionSpeed) + " E" + StrF(-retractionDistance) + "\n")
 }
 
-func pathesToGCode(pths []Path, comment string, feedrate int, b *bytes.Buffer) {
-	eOff := 0.0 //TODO: fix extruder value
+func pathesToGCode(pths []Path, comment string, feedrate int, extParams ExtrusionParams, eOff float64, b *bytes.Buffer) float64 {
+
 	b.WriteString(";" + comment + "\n")
+
+	// Set the printing speed for this path
+	printSpeedToGCode(feedrate, b)
+
 	for _, p := range pths {
 		// Retraction first
 		retractionToGCode(b, p.Retraction, p.RetractionDistance, p.RetractionSpeed)
 
-		// Set the printing speed for this path
-		printSpeedToGCode(feedrate, b)
```

Comment:
A retraction speed could be different to printing speed, therefore we need to set the printing speed for every path.

Reply:
According to gcode documentation(https://marlinfw.org/docs/gcode/G000-G001.html) and our code I think that during retraction we changed speed only executed command.
`b.WriteString("; Retraction\n")
b.WriteString("G1 F" + strconv.Itoa(retractionSpeed) + " E" + StrF(-retractionDistance) + "\n")`

 See examples in gcode documentation.